chatkit-core/src/main/kotlin/com/pusher/chatkit/ChatkitTokenProvider.kt

```diff
@@ -10,135 +10,143 @@ import com.pusher.util.flatten
-                        .url(httpUrl)
-                        .post(requestBody(tokenParams))
-                        .build()
-                val response = client.newCall(request).execute()
-
-                if (response.isSuccessful && response.code() in 200..299) {
-                    parseTokenResponse(response)
-                } else {
-                    response.asError().asFailure()
-                }
+    private fun fetchTokenFromEndpoint(): Future<Result<String, Error>> =
+        Futures.schedule {
+            val request = Request.Builder()
+                .url(httpUrl)
+                .post(requestBody())
+                .build()
+            val response = client.newCall(request).execute()
+
+            if (response.isSuccessful && response.code() in 200..299) {
+                parseTokenResponse(response)
+            } else {
+                response.asError().asFailure()
             }
+        }
 
-    private fun requestBody(tokenParams: Any?) = FormBody.Builder().apply {
+    private fun requestBody() = FormBody.Builder().apply {
         add("grant_type", "client_credentials")
-        add(authData)
-        if (tokenParams is ChatkitTokenParams) add(tokenParams.extras)
+        add(authData())
     }.build()
 
-    private fun FormBody.Builder.add(map: Map<String, String>) = {
+    private fun FormBody.Builder.add(map: Map<String, String>) {
         for ((k, v) in map) {
             add(k, v)
         }
     }
 
     private fun Response.asError(): Error = Errors.response(
-            statusCode = code(),
-            headers = headers().toMultimap(),
-            error = body()?.string() ?: ""
+        statusCode = code(),
+        headers = headers().toMultimap(),
+        error = body()?.string() ?: ""
     )
 
     private fun parseTokenResponse(response: Response): Result<String, Error> {
         return response.body()
-                ?.string()
-                ?.parseAs<TokenResponse>()
-                .orElse { Errors.network("Could not parse token from response: $response") }
-                .flatten()
-                .map { token ->
-                    tokenCache.cache(token.accessToken, token.expiresIn.toLong())
-                    token.accessToken
-                }
+            ?.string()
+            ?.parseAs<TokenResponse>()
+            .orElse { Errors.network("Could not parse token from response: $response") }
+            .flatten()
+            .map { token ->
+                tokenCache.cache(token.accessToken, token.expiresIn.toLong())
+                token.accessToken
+            }
     }
 }
 
-data class TokenResponse(
+private data class TokenResponse(
     val accessToken: String,
     val tokenType: String,
     val expiresIn: String,
     val refreshToken: String
 )
 
-data class ChatkitTokenParams(
-    val extras: Map<String, String> = emptyMap()
-)
-
 /**
- * Default token expiry tolerance - 10 minutes
- * */
+ * Default token expiry tolerance — 10 minutes.
```

Comment:
I'm not sure i like the minus symbol here, can we just use the word `is` instead?

Reply:
It's existing docs, indeed I changed it from minus to em dash which is typographically correct. Still if the dash didn't look good to you I [changed it to "of"](https://github.com/pusher/chatkit-android/commit/9844e12e91e1748deee82cf61082e108a084c852) as "is" didn't fit well definition (of what it is, not what is the value of it), hope it's fine now?

Reply:
Yeah, reads fine now :) Thanks!